torch/csrc/jit/passes/quantization.cpp

```diff
@@ -564,13 +587,7 @@ void QuantizeHelper::quantizeTensor(Value* v) {
   // NB: v is updated here, since removeObserver replaces
   // v with the input to the observer call
   v = removeObserver(v, observer_name.value());
-  Node* dequant;
-  dequant = insertQuantDeQuantCall(v, qparams, scalar_type);
-  v->replaceAllUsesWith(dequant->output());
-  Node* q = dequant->input(0)->node();
-  // replaceAllUsesWith rewrote all uses of V, but we want to keep one: the one
-  // used in quant node. Restore it here:
-  q->replaceInputWith(dequant->output(), v);
+  insertQuantDeQuantCall(v, qparams, scalar_type, observer_name.value());
```

Comment:
I like how this place got cleaned up, but maybe we also want to revisit how we split logic between `removeObserver` and `insertQuantDeQuantCall`? It might make sense to merge them into one, since `removeObserver` currently doesn't remove it completely (it keeps the getAttr node). By the way, we should probably remove that getAttr node in the end too.

Reply:
Yes, I think the latest removeObserver code also removes the getAttr node.